jdbc/src/main/java/io/kestra/jdbc/runner/JdbcExecutor.java

```diff
@@ -196,6 +205,36 @@ public void run() {
                 );
             }
         );
+
+    }
+
+    @Scheduled(fixedDelay = "${kestra.heartbeat.frequency}")
+    protected void workersUpdate() {
+        workerInstanceRepository.heartbeatsStatusUpdate();
+        if (workerInstanceRepository.heartbeatsCleanup()) {
+            this.deadWorkerTaskResubmit();
+        }
+    }
+
+    private void deadWorkerTaskResubmit() {
+        workerJobRunningRepository.getWorkerJobWithWorkerDead(workerInstanceRepository.findAllAlive().stream().map(workerInstance -> workerInstance.getWorkerUuid().toString()).collect(Collectors.toList()))
```

Comment:
Please gather the list of workerInstance before.
I had a hard time understanding the code (and understood it wrong the first time).